apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/status/AnnotationMatchExceptionCheckStrategy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context.status;
+
+import java.lang.annotation.Annotation;
+import java.util.HashSet;
+import java.util.Objects;
+import org.apache.skywalking.apm.agent.core.boot.ServiceManager;
+
+public class AnnotationMatchExceptionCheckStrategy implements ExceptionCheckStrategy {
+
+    private final HashSet<Class<? extends Throwable>> ignoredExceptions = new HashSet<>(16);
+
+    @Override
+    public boolean isError(final Throwable e) {
+        if (ignoredExceptions.contains(e.getClass())) {
+            return false;
+        }
+        StatusCheckService statusTriggerService = ServiceManager.INSTANCE.findService(StatusCheckService.class);
+        Class<? extends Annotation> ignoredExceptionAnnotationClass = statusTriggerService.ignoredExceptionAnnotationClass;
+        if (Objects.isNull(ignoredExceptionAnnotationClass)) {
+            return true;
+        }
+        if (e.getClass().isAnnotationPresent(ignoredExceptionAnnotationClass)) {
```

Comment:
I still don't like the idea use the annotation. 
1. The toolkit classes are in the user class loader zone, you can't assume it is usable in the agent core.
1. Why simply use the exception name string match? This is much safer and easier than this.
1. Like (2), the exception class could be cached.

Reply:
> 1. The toolkit classes are in the user class loader zone, you can't assume it is usable in the agent core.
1. load the ignoredExceptionAnnotationClass by application classloader, it is already passed in the manual test.


> 2\. Why simply use the exception name string match? This is much safer and easier than this.


If i was a user,i would to add ignore exception by annotation,this would be easy to control rather than change env config frequency. in our department,skywalking is throught as a platform, env is a only way to custom config.But env config is out of user code config, user would not like to frequency config.Env should keep stable.






> 3\. Like (2), the exception class could be cached.

private final HashSet<Class<? extends Throwable>> ignoredExceptions = new HashSet<>(16);
already cache ignored exceptions.
we should add a new cache to store unignored exceptions.Is that what you're saying?

Reply:
> load the ignoredExceptionAnnotationClass by application classloader, it is already passed in the manual test.

`ignoredExceptionAnnotationClass by application classloader`, it could not be.

> private final HashSet<Class<? extends Throwable>> ignoredExceptions = new HashSet<>(16);
already cache ignored exceptions.
we should add a new cache to store unignored exceptions.Is that what you're saying?

We could use a Map directly.

Reply:
Also, the key is not right, you need to use `classLoaderInstance + Exception class` as the key. The same class could be loaded by different class loaders with the same name.